tests/unit/test_endpoint_provider.py

```diff
@@ -343,3 +344,24 @@ def test_ruleset_unknown_parameter_type_raises(partitions):
             partitions=partitions,
         )
     assert "Unknown parameter type: list." in str(exc_info.value)
+
+
+@pytest.fixture(scope="class")
+def test_endpoint_provider(endpoint_provider, request):
+    request.cls.endpoint_provider = endpoint_provider
+
+
+@pytest.mark.usefixtures("test_endpoint_provider")
+class EndpointProviderCacheTestCase(unittest.TestCase):
```

Comment:
I think the reasoning behind this was because I was mixing/matching pytest and unittest. I see that pytest has the `monkeypatch` fixture. I'll try to incorporate that instead here.